cartographer/common/optional.h

```diff
@@ -0,0 +1,39 @@
+#ifndef CARTOGRAPHER_COMMON_OPTIONAL_H_
```

Comment:
This needs a license comment. Same with the other file.

Reply:
done